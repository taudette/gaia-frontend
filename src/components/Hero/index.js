import React from 'react';
import styles from '../../styles/components/_hero.scss'

const HeroComponent = (props) => {
  return (
    <section >
      <div className={styles.heroContainer}>
        <img src={props.hero.lgImage} alt="hero-image" />
        <div className={styles.text}>
          <h2>{props.hero.name}</h2>
          <p className="hidden-xs"><small>{props.hero.body}</small></p>
        </div>
      </div>
    </section>
  )
}

export default HeroComponent


