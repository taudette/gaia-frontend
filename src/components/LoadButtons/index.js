import React from 'react';
import styles from '../../styles/components/_loadButtons.scss'
import { Row, Col } from 'react-bootstrap'

const LoadButtonsComponent = (props) => {
  return (
    <Row className={styles.buttonRow}>
      {/* don't show load more button if no more to load*/}
      { props.endIndex != props.listLength &&
        <Col smOffset={4} sm={4} xs={12} className="text-center">
          <button className={styles.loadButton} onClick={props.addVideos}>Load More</button>
        </Col>
      }
      <Col sm={4} xs={12} className="text-right">
        <button onClick={props.goToTop} className={styles.upButton}><i className={styles.arrow}></i></button>
      </Col>
    </Row>
  )
}

export default LoadButtonsComponent