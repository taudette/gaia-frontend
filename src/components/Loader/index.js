import React from 'react';
import loader from '../../styles/components/_loader.scss';

const LoaderComponent = () => (
  <div className={loader.loaderContainer}>
    <div className={loader.loader}></div>
  </div>
)

export default LoaderComponent