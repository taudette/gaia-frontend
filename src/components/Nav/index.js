import React from 'react';

export const NavComponent = () => (
  <div>
    Nav Component
  </div>
)

export default NavComponent
