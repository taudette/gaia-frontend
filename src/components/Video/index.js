import React from 'react';
import styles from '../../styles/components/_video.scss'

export const VideoComponent = (props) => {
  console.log(props)
  return (
      <div>
        {/* only render video if image and title are avaiable, check the rest individually */}
        { props.img && props.title && 
          <div className={styles.tileContainer}>
            <img src={props.img} alt="video-image" /> 
            <p className="">{props.title}</p> 
            { props.segment && <p>segment: {props.segment}</p> }
            { props.likes && <p>likes: {props.likes}</p> }
            { props.dislikes && <p>dislikes: {props.dislikes}</p> }
            { props.seasons && <p>seasons: {props.seasons} </p> } 
            { props.episodes && <p>episodes: {props.episodes} </p> } 
          </div>
        }
      </div>
      
  )
}

export default VideoComponent
