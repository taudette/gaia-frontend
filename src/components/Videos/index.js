import React from 'react';
import VideoComponent from '../Video'
import styles from '../../styles/components/_videos.scss';

export const VideosComponent = (props) => {
  if (props.loading) {
    return <p>Loading...</p>
  }

  return (
    <div >
      {props.videos &&
        <ul className={styles.videoList}>
          {props.videos.map((video, index) => {
            return <VideoComponent {...video} key={index} />
          })}
        </ul>
      }
    </div>
  )
}

export default VideosComponent
