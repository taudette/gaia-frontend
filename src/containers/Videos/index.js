import React from 'react';
import VideosComponent from '../../components/Videos'

export const VideosContainer = () => (
  <div>
    Videos Container
    <VideosComponent />
  </div>
)

export default VideosContainer
