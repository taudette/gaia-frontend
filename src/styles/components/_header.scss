@import '../base/_settings.scss';

header {
  padding-left: 4rem;
  padding-right: 4rem;
  display: flex;
  align-items: center;
  height: 5rem;
}

.row {
  width: 100%;
}

.logoContainer {
  width: 7rem;
}

.userContainer {
  float: right;
  text-align: right;

  img {
    width: 2rem;
    margin-left: 1rem;
  }

  span {
    color: darkgrey;
  }
}

.icon {
  background-image: url("../../../public/images/header-logo.svg");
}

