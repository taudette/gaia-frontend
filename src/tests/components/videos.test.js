import React from 'react';
import { shallow } from 'enzyme';
import { VideosComponent } from '../../components/Videos';
//import videos from '../fixtures/videos';

test('should render videos component without videos', () => {
  const wrapper = shallow(<VideosComponent videos={[]} loading={false} />)
  expect(wrapper).toMatchSnapshot();
})