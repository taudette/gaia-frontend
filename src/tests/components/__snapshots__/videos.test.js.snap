// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render videos 1`] = `
<section>
  <div
    className="videoList"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="buttonCol hidden-xs"
        componentClass="div"
        lg={3}
        lgOffset={9}
        md={4}
        mdOffset={8}
        sm={6}
        smOffset={6}
        xs={12}
      >
        <p>
          SORT BY
        </p>
        <DropdownButton
          bsStyle="default"
          className="dropDownButton"
          id="sort videos"
          title="Choose an Option"
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="1"
            header={false}
          >
            Name
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="2"
            header={false}
          >
            Date
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="2"
            header={false}
          >
            Category
          </MenuItem>
        </DropdownButton>
      </Col>
    </Row>
    <p>
      We suggest you start here →
    </p>
    <ul>
      <VideoComponent
        dislikes={48}
        episodes=""
        img="https://img.gtvcdn.com/cdn/farfuture/29uIix8RhExtaI54apLJdD1J62mat_Cxx6mMQNxxUek/mtime%3A1496214668/sites/default/files/imagecache/hero_320x200/img_16x9_landsacpe_title/142666_wt_s26e5_the-mystery-of-comet-shoemaker-levy-9_16x9.jpg"
        key="0"
        likes={888}
        seasons=""
        segment="Seeking Truth"
        title="[#220] The Mystery of Comet Shoemaker-Levy 9"
      />
      <VideoComponent
        dislikes={46}
        episodes=""
        img="https://img.gtvcdn.com/cdn/farfuture/q-Dlko7q-kZVT6UZhsznASzrDxjOgjoDVBUg2HmpvPg/mtime%3A1486967204/sites/default/files/imagecache/hero_320x200/img_16x9_landsacpe_title/137021_wt_s24e14_revealing-the-cosmic-quantum-medium_part-2_16x9_0.jpg"
        key="1"
        likes={576}
        seasons=""
        segment="Seeking Truth"
        title="[#206] Revealing the Cosmic Quantum Medium Part 2"
      />
      <VideoComponent
        dislikes={40}
        episodes=""
        img="https://img.gtvcdn.com/cdn/farfuture/gpoPETTfRHyBACdDoyH7IP_Q5M29ZtBk6KssnJEZr8U/mtime%3A1489392081/sites/default/files/imagecache/hero_320x200/img_16x9_landsacpe_title/138171_wt_s24e16_sacred-geometry-of-the-squared-circle_16x9_0.jpg"
        key="2"
        likes={718}
        seasons=""
        segment="Seeking Truth"
        title="[#208] Sacred Geometry of the Squared Circle"
      />
      <VideoComponent
        dislikes={94}
        episodes=""
        img="https://img.gtvcdn.com/cdn/farfuture/aWjV45VC818oeNRpQ0WpDtOyUhHzd-QttMgyb2Su2QA/mtime%3A1492622348/sites/default/files/imagecache/hero_320x200/img_16x9_landsacpe_title/140516_om_s8e19_alice-bailey-and-the-circle-of-initiates_w_klaire-roy_16x9.jpg"
        key="3"
        likes={522}
        seasons=""
        segment="Seeking Truth"
        title="Alice Bailey and the Circle of Initiates with Klaire Roy"
      />
    </ul>
  </div>
</section>
`;

exports[`should render videos component without videos 1`] = `
<section>
  <div
    className="videoList"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="buttonCol hidden-xs"
        componentClass="div"
        lg={3}
        lgOffset={9}
        md={4}
        mdOffset={8}
        sm={6}
        smOffset={6}
        xs={12}
      >
        <p>
          SORT BY
        </p>
        <DropdownButton
          bsStyle="default"
          className="dropDownButton"
          id="sort videos"
          title="Choose an Option"
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="1"
            header={false}
          >
            Name
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="2"
            header={false}
          >
            Date
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey="2"
            header={false}
          >
            Category
          </MenuItem>
        </DropdownButton>
      </Col>
    </Row>
    <p>
      We suggest you start here →
    </p>
    <ul />
  </div>
</section>
`;
